rasa/train.py

```diff
@@ -683,20 +693,37 @@ async def _train_nlu_with_validated_data(
         return _train_path
 
 
-def _nlu_model_for_finetuning(
+async def _nlu_model_for_finetuning(
     model_to_finetune: Text,
-    new_config: Dict[Text, Any],
+    file_importer: TrainingDataImporter,
     finetuning_epoch_fraction: float = 1.0,
+    called_from_combined_training: bool = False,
 ) -> Optional[Interpreter]:
+
     path_to_archive = model.get_model_for_finetuning(model_to_finetune)
     if not path_to_archive:
         return None
 
     with model.unpack_model(path_to_archive) as unpacked:
         _, old_nlu = model.get_model_subdirectories(unpacked)
+        new_fingerprint = await model.model_fingerprint(file_importer)
+        old_fingerprint = model.fingerprint_from_path(unpacked)
+        if not model.can_finetune(
+            old_fingerprint,
+            new_fingerprint,
+            nlu=True,
+            core=called_from_combined_training,
+        ):
+            rasa.shared.utils.cli.print_error_and_exit(
+                "NLU model can not be finetuned."
```

Comment:
Is it possible to point to the possible reason? I think we can definitely distinguish between the three cases right?

Reply:
Yes, would required a bit of refactoring - but definitely possible.

Reply:
If it's major refactoring, we could also create an issue and tackle it after the release.

Reply:
Actually it will currently log it to `info` (in `did_section_fingerprint_change`). Maybe that is enough for now?

Reply:
Tried it out. The explanation is not very helpful but I'd say it's enough for now. Can you create an enhancement issue to make these logs more verbose so that the user understands exactly where the problem lies?